cdap-app-fabric/src/main/java/io/cdap/cdap/security/impersonation/DefaultUGIProvider.java

```diff
@@ -172,4 +202,59 @@ private File localizeKeytab(Location keytabLocation) throws IOException {
 
     return localKeytabFile.toFile();
   }
+
+  /**
+   * Returns the runtime user supplied arguments to a program if entity is of type ProgramRun
+   *
+   * @param programId the entity id
+   * @return properties map
+   */
+  private Map<String, String> getRuntimeProperties(NamespacedEntityId programId) {
+    if (programId instanceof ProgramRunId) {
+      ProgramRunId runId = ((ProgramRunId) programId);
+      RunRecordMeta runRecord = null;
+
+      long sleepDelayMs = TimeUnit.MILLISECONDS.toMillis(100);
+      long timeoutMs = TimeUnit.SECONDS.toMillis(10);
+
+      try {
+        runRecord = Retries.callWithRetries(() -> {
+          try {
+            RunRecordMeta rec = store.getRun(runId);
+            if (rec != null) {
+              return rec;
+            }
+          } catch (Exception e) {
+            LOG.debug("Exception while trying to fecth RunRecord %s, retrying again: ", e.getMessage());
+          }
+          throw new Exception("Retrying again to fetch run record");
+        }, RetryStrategies.timeLimit(timeoutMs, TimeUnit.MILLISECONDS,
+                  RetryStrategies.fixDelay(sleepDelayMs, TimeUnit.MILLISECONDS)),
+                  Exception.class::isInstance);
+      } catch (Exception e1) {
+      }
+
+      if (runRecord == null) {
+        LOG.debug("Could not obtain program's meta run record");
+        return Collections.emptyMap();
+      }
+
+      Gson gson = new Gson();
+      Type stringStringMap = new TypeToken<Map<String, String>>() { }.getType();
+
+      Map<String, String> properties = runRecord.getProperties();
+      String runtimeArgsJson = properties.get("runtimeArgs");
+      if (runtimeArgsJson == null) {
+        LOG.debug("Could not find any runtime args");
+        return Collections.emptyMap();
+      }
+
+      properties = gson.fromJson(runtimeArgsJson, stringStringMap);
```

Comment:
Assignment is not needed. Just do `return gson.fromJson(runtimeArgsJson, stringStringMap);`

Reply:
Done. Thanks